h2o-py/h2o/estimators/estimator_base.py

```diff
@@ -57,6 +58,13 @@ class H2OEstimator(ModelBase):
     """
     
     supervised_learning = None  # overridden in implementation
+    __default_params = None     # mangled/hidden: use `_default_param_value` to access default param values internally.
+
+    def __new__(cls, *args, **kwargs):
+        if cls.__default_params is None:
+            cls.__default_params = object()  # anything but None to avoid infinite recursion
```

Comment:
Would it be easier to create the reference instance on demand in `_check_and_save_parm`?

Reply:
yes, I succumbed to excessive refinement… will improve